src/gax-internal/src/observability.rs

```diff
@@ -0,0 +1,275 @@
+// Copyright 2025 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::options::InstrumentationClientInfo;
+use gax::options::RequestOptions;
+
+/// Holds information extracted from HTTP requests and responses,
+/// formatted to align with OpenTelemetry semantic conventions for tracing.
+/// This struct is used to populate attributes on tracing spans.
+#[derive(Debug, Clone)]
+#[allow(dead_code)] // TODO(#3239): Remove once used in http.rs
+pub(crate) struct HttpSpanInfo {
+    // Attributes for OpenTelemetry SDK interop
+    otel_kind: String,   // "Client"
+    otel_name: String,   // "{METHOD} {url.template}" or "{METHOD}"
+    otel_status: String, // "Unset", "Ok", "Error"
```

Comment:
Even if these elements `#[doc(hidden)]` I think just using standard Rust documentation makes the code easier to read (and we could generate "private") documentation for ourselves:


```suggestion
    /// The type of request, typically `Client`.
    otel_kind: String,
    /// The request name, either "{METHOD} {url.template}" or "{METHOD}"
    otel_name: String,
    /// The result of the request, one of: "Unset", "Ok", "Error"
    otel_status: String,
```

Reply:
No, you don't need to add any directives.

All of the modules are feature gated:

https://github.com/googleapis/google-cloud-rust/blob/main/src/gax-internal/src/lib.rs

And we tell docs.rs to build without any features enabled:

https://github.com/googleapis/google-cloud-rust/blob/0431b38688f12fd7a230bb0a923091a875f652fe/src/gax-internal/Cargo.toml#L29-L32